lib/URLButton.js

```diff
@@ -0,0 +1,130 @@
+//reimplimentation of add-on sdk button/toggle, found at
+//https://dxr.mozilla.org/mozilla-central/source/addon-sdk/source/lib/sdk/ui/button/toggle.js
```

Comment:
Links like this will break, in time. Better to use https://dxr.mozilla.org/mozilla-central/rev/fdcee57b4e4f66a82831ab01e61500da98a858e8/addon-sdk/source/lib/sdk/ui/button/toggle.js, which you get from the permalink button on the right side of DXR.